File: app/src/main/java/com/apicloud/pkg/sdk/component/AppComponent.java
package com.apicloud.pkg.sdk.component;

import com.apicloud.pkg.sdk.module.AppModule;
import com.apicloud.pkg.sdk.module.ApiModule;

import dagger.Component;

import android.content.Context;

import com.apicloud.pkg.sdk.api.AppApi;

@Component(modules = {AppModule.class, ApiModule.class})
public interface AppComponent {
    Context getContext();

    AppApi getShopApi();
}

File: app/src/main/java/com/apicloud/pkg/sdk/component/AppComponent.kt
package com.apicloud.pkg.sdk.component

import android.content.Context
import com.apicloud.pkg.sdk.api.AppApi
import com.apicloud.pkg.sdk.module.ApiModule
import com.apicloud.pkg.sdk.module.AppModule
import dagger.Component

@Component(modules = arrayOf(AppModule::class, ApiModule::class))
interface AppComponent {
    val context: Context

    val shopApi: AppApi
}